contracts/marketplace/CruzoMarket.sol

```diff
@@ -67,7 +71,14 @@ contract CruzoMarket is ERC1155Holder, Ownable {
         tradeCounter += 1;
         emit TradeStatusChange(tradeCounter - 1, "Open");
     }
-
+    function existTrade(uint256 _itemId) public view returns (bool){
+        for (uint i; i <= tradeCounter; i++) {
```

Comment:
The provided solution is not optimal from gas usage perspective. We would suggest to use a mapping with a complex key pattern to store our trades. This way we can have O(1) complexity which is much better than for-loop approach. Could you re-work please?